src/data_readers/data_reader_jag_conduit.cpp

```diff
@@ -310,17 +310,46 @@ const conduit::Node& data_reader_jag_conduit::get_conduit_node(const conduit::No
 }
 
 bool data_reader_jag_conduit::load_conduit_node(const size_t i, const std::string& key, conduit::Node& node) const {
+
+  if (m_io_thread_pool != nullptr && m_using_random_node.count(m_io_thread_pool->get_local_thread_id())) {
+    LBANN_ERROR("previously retrieved a random conduit node from data_store, so shouldn't be here");
+  }
+
   const sample_t& s = m_sample_list[i];
   const std::string& sample_name = s.second;
   const std::string path = sample_name + key;
 
   sample_file_id_t id = s.first;
   hid_t h = m_sample_list.get_samples_hdf5_handle(id);
   if (h <= static_cast<hid_t>(0) || !conduit::relay::io::hdf5_has_path(h, path)) {
-    const std::string& file_name = m_sample_list.get_samples_filename(id);
-    LBANN_ERROR(get_type() + ":: Cannot open file " + file_name + \
-                " for sample "+ sample_name);
-    return false;
+    if (m_data_store != nullptr) {
+      const conduit::Node obj = m_jag_store->get_random_node();
+      node = obj["data"];
+      const std::vector<std::string>& child_names = node.child_names();
+      const std::string cur_child = child_names[0];
+      const std::string new_child = pad(std::to_string(i), SAMPLE_ID_PAD, '0');
+      node.rename_child(cur_child, new_child);
+      const std::string& file_name = m_sample_list.get_samples_filename(id);
```

Comment:
I think that you need something like `m_using_random_node.emplace(m_io_thread_pool->get_local_thread_id());`